From 3af51f98fa900bb210979321bfaffeb8ba5b1649 Mon Sep 17 00:00:00 2001
Message-Id: <3af51f98fa900bb210979321bfaffeb8ba5b1649.1368098699.git.minovotn@redhat.com>
In-Reply-To: <618a4b91ddb04b21f9dc0c1defe7693fb7cc1748.1368098699.git.minovotn@redhat.com>
References: <618a4b91ddb04b21f9dc0c1defe7693fb7cc1748.1368098699.git.minovotn@redhat.com>
From: Bandan Das <bsd@redhat.com>
Date: Tue, 7 May 2013 04:45:53 +0200
Subject: [PATCH 21/24] pc: Add a new machine type : rhel6.5.0

RH-Author: Bandan Das <bsd@redhat.com>
Message-id: <1367901955-8217-2-git-send-email-bsd@redhat.com>
Patchwork-id: 51162
O-Subject: [PATCH RHEL-6.5 qemu-kvm 1/3 v2] pc: Add a new machine type : rhel6.5.0
Bugzilla: 821741
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>

Signed-off-by: Bandan Das <bsd@redhat.com>
---
 hw/pc.c | 24 ++++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/pc.c | 24 ++++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

diff --git a/hw/pc.c b/hw/pc.c
index 29e5e73..526369c 100644
--- a/hw/pc.c
+++ b/hw/pc.c
@@ -1751,6 +1751,27 @@ static void pc_rhel620_compat(void)
     set_pmu_passthrough(false);
 }
 
+static void pc_init_rhel650(ram_addr_t ram_size,
+			    const char *boot_device,
+			    const char *kernel_filename,
+			    const char *kernel_cmdline,
+			    const char *initrd_filename,
+			    const char *cpu_model)
+{
+	rhel_common_init("RHEL 6.5.0 PC", 0);
+	pc_init_pci(ram_size, boot_device, kernel_filename, kernel_cmdline,
+		    initrd_filename, setdef_cpu_model(cpu_model, "cpu64-rhel6"));
+}
+
+static QEMUMachine pc_machine_rhel650 = {
+	.name = "rhel6.5.0",
+	.alias = "pc",
+	.desc = "RHEL 6.5.0 PC",
+	.init = pc_init_rhel650,
+	.max_cpus = 255,
+	.is_default = 1,
+};
+
 static void pc_init_rhel640(ram_addr_t ram_size,
                             const char *boot_device,
                             const char *kernel_filename,
@@ -1765,11 +1786,9 @@ static void pc_init_rhel640(ram_addr_t ram_size,
 
 static QEMUMachine pc_machine_rhel640 = {
     .name = "rhel6.4.0",
-    .alias = "pc",
     .desc = "RHEL 6.4.0 PC",
     .init = pc_init_rhel640,
     .max_cpus = 255,
-    .is_default = 1,
 };
 
 static void pc_init_rhel630(ram_addr_t ram_size,
@@ -1980,6 +1999,7 @@ static QEMUMachine pc_machine_rhel540 = {
 
 static void rhel_machine_init(void)
 {
+    qemu_register_machine(&pc_machine_rhel650);
     qemu_register_machine(&pc_machine_rhel640);
     qemu_register_machine(&pc_machine_rhel630);
     qemu_register_machine(&pc_machine_rhel620);
-- 
1.7.11.7

